### PR TITLE
Redirect list convert into a view, exposed filters added.

### PR DESCRIPTION
**Pull request by giancarlosotelo**
Added the view and also added a formatter to Redirect Source item.

### CODE REVIEW

File: redirect.views.inc [hunk 1]
```diff
@@ -8,7 +8,7 @@
 /**
  * Implements hook_views_data().
  */
-function redirect_views_data() {
+function redirect_testing_views_data() {
```
Comment (Berdir) [inline redirect.views.inc]: This needs to be removed now. (the whole function)

File: src/Plugin/Field/FieldFormatter/RedirectSourceFormatter.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items) {
+    $elements = array();
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = array(
+        '#markup' => nl2br(SafeMarkup::checkPlain($item->path))
```
Comment (Berdir) [inline src/Plugin/Field/FieldFormatter/RedirectSourceFormatter.php]: I think we don't need the nl2br() and Safemarkup here.

File: src/Entity/Redirect.php [hunk 3]
```diff
@@ -30,6 +30,7 @@
  *       "delete" = "Drupal\redirect\Form\RedirectDeleteForm",
  *       "edit" = "Drupal\redirect\Form\RedirectForm"
  *     },
+ *     "views_data" = "Drupal\views\EntityViewsData",
```
Comment (Berdir) [inline src/Entity/Redirect.php:33]: There is a custom list builder here for our list. Now that we have a view, we can drop that. Remove our custom class but keep the default EntityListBuilder (for operations support), similar to the default EntityViewsData.

Also, we need to depend on the views module in the info.yml file then. That will also start with addressing those test fails.

File: src/Tests/RedirectUITest.php [hunk 4]
```diff
@@ -166,13 +166,33 @@ public function testRedirectUI() {
...
+    $this->assertTrue(count($rows) == 1);
+
     // Finally test the delete action.
     $this->drupalGet('admin/config/search/redirect');
     $this->clickLink(t('Delete'));
     $this->assertRaw(t('Are you sure you want to delete the URL redirect from %source to %redirect?', array('%source' => '/non-existing?key=value', '%redirect' => '/node')));
     $this->drupalPostForm(NULL, array(), t('Delete'));
     $this->assertUrl('admin/config/search/redirect');
-    $this->assertText(t('There is no @label yet.', array('@label' => 'Redirect')));
+    $this->assertText(t('There is no @label yet.', array('@label' => 'redirect')));
```
Comment (Berdir) [inline src/Tests/RedirectUITest.php]: Remove the label placeholder, just add redirect directly in the assert message.

File: src/Tests/RedirectUITest.php [hunk 5]
```diff
@@ -166,13 +166,33 @@ public function testRedirectUI() {
     ), t('Save'));
     $this->assertRaw(t('The anchor fragments are not allowed.'));
 
+    // Test filters.
+    // Filter  with other value.
+    $this->drupalGet('admin/config/search/redirect', array('query' => array(
+      'status_code' => '401',
```
Comment (Berdir) [inline src/Tests/RedirectUITest.php]: use a 3xx status code, only those are valid values and we'll eventually limit it to that.

**Comment by Berdir** [comment]
Needs to merged with 8.x-1.x. There is a new created column now that needs to be added to the view and it will conflict on the removed list builder.

**Comment by giancarlosotelo** [comment]
tests are not working because a patch about LanguageFormatter is needed, https://www.drupal.org/node/2461013.

**Comment by Berdir** [comment]
This looks pretty good!

from/to filter doesn't seem to be working, though, might need some custom views plugins.

Also ,in the table settings, we should allow sorting for all columns where it makes sense and sort by created desc by default.

**Comment by giancarlosotelo** [comment]
I was adding a custom view plugin but I realized that changing the operator in the filter criteria was enough. I added a test also for both fields.

**Comment by Berdir** [comment]
Indeed, this works well well now.

Just some cleanup and we're done. Drop redirect.views.inc completely and drop the remaining custom views plugins, we no longer need those.

**Comment by Berdir** [comment]
Very nice, merged!

**Pull request closed without merging**